k8s/k8s.go

```diff
@@ -294,6 +296,69 @@ func CmdAddK8s(args *skel.CmdArgs, conf utils.NetConf, nodename string, calicoCl
 	return result, nil
 }
 
+// CmdDelK8s performs the "DEL" operation on a kubernetes pod.
+// The following logic only applies to kubernetes since it sends multiple DELs for the same
+// endpoint. See: https://github.com/kubernetes/kubernetes/issues/44100
+// We store CNI_CONTAINERID as ActiveInstanceID in WEP Metadata for k8s,
+// so in this function we need to get the WEP and make sure we check if ContainerID and ActiveInstanceID
+// are the same before deleting the pod being deleted.
+func CmdDelK8s(c *calicoclient.Client, ep api.WorkloadEndpointMetadata, args *skel.CmdArgs, conf utils.NetConf, logger *log.Entry) error {
+	wep, err := c.WorkloadEndpoints().Get(ep)
+	if err != nil {
+		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+			// We can talk to the datastore but WEP doesn't exist in there,
+			// but we still want to go ahead with the clean up. So, log a warning and continue with the clean up.
+			logger.WithField("WorkloadEndpoint", ep).Warning("WorkloadEndpoint does not exist in the datastore, moving forward with the clean up")
+		} else {
+			// Could not connect to datastore (connection refused, unauthorized, etc.)
+			// so we have no way of knowing/checking ActiveInstanceID. To protect the endpoint
+			// from false DEL, we return the error without deleting/cleaning up.
+			return err
+		}
+
+		// Check if ActiveInstanceID is populated (it will be an empty string "" if it was populated
+		// before this field was added to the API), and if it is there then compare it with ContainerID
+		// passed by the orchestrator to make sure they are the same, return without deleting if they aren't.
+	} else if wep.Metadata.ActiveInstanceID != "" && args.ContainerID != wep.Metadata.ActiveInstanceID {
+		logger.WithField("WorkloadEndpoint", wep).Warning("CNI_ContainerID does not match WorkloadEndpoint ActiveInstanceID so ignoring the DELETE cmd.")
+		return nil
+	}
+
+	// Delete the WorkloadEndpoint object from the datastore.
+	// In case of k8s, where we are deleting the WEP we got from the Datastore,
+	// this Delete is a Compare-and-Delete, so if *any* field in the WEP changed from
+	// the time we get WEP until here then the Delete operation will fail.
+	if err = c.WorkloadEndpoints().Delete(wep.Metadata); err != nil {
```

Comment:
I added an else there so it only tries to delete when get succeeds (forgot to save and commit it)